modules/configuration/src/main/java/org/apache/ignite/internal/configuration/tree/NamedListNode.java

```diff
@@ -489,9 +489,26 @@ public void construct(String key, ConfigurationSource src, boolean includeIntern
 
         newElement.setInjectedNameFieldValue(key);
 
+        newElement.internalId(generateInternalId());
+
         return new ElementDescriptor(newElement);
     }
 
+    /**
+     * Generates a new unique internal id for the node.
+     *
+     * @return Generated internal id.
+     */
+    private UUID generateInternalId() {
```

Comment:
Will it be regenerated for "pending" changes on every iteration of cfg apply process if invoke to ms failed?

Reply:
It will be re-generated for every retry attempt